javascript/script.js

```diff
@@ -0,0 +1,90 @@
+const statusDisplay = document.querySelector('#score');
+
+let gameActive = true;
+let currentPlayer = "X";
+let gameState = ["", "", "", "", "", "", "", "", ""];
+const winningMessage = () => `Player ${currentPlayer} has won!`;
+const drawMessage = () => `Game ended in a draw!`;
+const currentPlayerTurn = () => `It's ${currentPlayer}'s turn`;
+
+
+const winningConditions = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6]
+];
+
+statusDisplay.innerHTML = currentPlayerTurn();
+
+const handleCellClick = (clickedCellEvent) => {
+    const clickedCell = clickedCellEvent.target;
+
+    const clickedCellIndex = parseInt(clickedCell.getAttribute('id'));
+    if (gameState[clickedCellIndex] !== "" || !gameActive) {
+        return;
+    }
+
+    handleCellPlayed(clickedCell, clickedCellIndex);
+    handleResultValidation();
+}
+
+
+const handleCellPlayed = (clickedCell, clickedCellIndex) => {
+    gameState[clickedCellIndex] = currentPlayer;
+    clickedCell.innerHTML = currentPlayer;
+}
+
+const handleResultValidation = () => {
+    let roundWon = false;
+    for (let i = 0; i <= 7; i++) {
+        const winCondition = winningConditions[i];
+        let a = gameState[winCondition[0]];
```

Comment:
It is not too hard to understand what's going on here, but consider using more descriptive variable names like firstSquare secondSquare thirdSquare